volatility3/framework/symbols/windows/extensions/consoles.py

```diff
@@ -0,0 +1,439 @@
+            0xF0,
+            0xA0,
+        )
+        if n == 0 and not valid:
+            print("Bad Dbcs Attribute {}".format(hex(c)))
+        return valid
+
+    def get_text(self, truncate=True):
+        """A convenience method to extract the text from the _ROW.  The _ROW
+        contains a pointer CharRow to an array of CharRowCell objects. Each
+        CharRowCell contains the wide character and an attribute. Enumerating
+        self.CharRow.Chars and casting each character to unicode takes too long,
+        so this reads the whole row into a buffer, then extracts the text characters."""
+
+        layer = self._context.layers[self.vol.layer_name]
+        offset = self.CharRow.Chars.vol.offset
+        length = self.RowLength * 3
+        char_row = layer.read(offset, length)
+        line = ""
+        try:
+            if char_row:
+                line = "".join(
+                    (
+                        char_row[i : i + 2].decode("utf-16le", errors="replace")
+                        if self._valid_dbcs(char_row[i + 2], char_row[i + 1])
+                        else ""
+                    )
+                    for i in range(0, len(char_row), 3)
+                )
+        except Exception as e:
+            print(e)
+            line = ""
+
+        if truncate:
+            return line.rstrip()
+        else:
+            return line
+
+
+class ALIAS(objects.StructType):
+    """An Alias Structure"""
+
+    def get_source(self):
+        if self.Source.Length < 8:
+            return self.Source.Chars.cast(
+                "string",
+                encoding="utf-16",
+                errors="replace",
+                max_length=self.Source.Length * 2,
+            )
+        elif self.Source.Length < 1024:
+            return self.Source.Pointer.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+
+    def get_target(self):
+        if self.Target.Length < 8:
+            return self.Target.Chars.cast(
+                "string",
+                encoding="utf-16",
+                errors="replace",
+                max_length=self.Target.Length * 2,
+            )
+        elif self.Target.Length < 1024:
+            return self.Target.Pointer.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+
+
+class EXE_ALIAS_LIST(objects.StructType):
+    """An Exe Alias List Structure"""
+
+    def get_exename(self):
+        exe_name = self.ExeName
+        # Windows 10 22000 and Server 20348 removed the Pointer
+        if isinstance(exe_name, objects.Pointer):
+            exe_name = exe_name.dereference()
+            return exe_name.get_string()
+
+        if self.ExeName.Length < 8:
+            return self.ExeName.Chars.cast(
+                "string",
+                encoding="utf-16",
+                errors="replace",
+                max_length=self.ExeName.Length * 2,
```

Comment:
Here `Length` is the length of the string, but below it's the length in bytes (because you check `< 1024` and then max it out at 512).  So... which is it?

Reply:
Honestly don't remember where the 1024 came from. It might just be to prevent something very large that's totally incorrect. I got rid of the 512 and used self.Length in both places now.